mlflow/server/js/src/experiment-tracking/components/RunView.js

```diff
@@ -51,6 +54,13 @@ export class RunViewImpl extends Component {
   componentDidMount() {
     const pageTitle = `${this.props.runDisplayName} - MLflow Run`;
     Utils.updatePageTitle(pageTitle);
+    // Request metric history for all the run metrics so that minimum
+    // and maximum values will be computed.
+    // Because the RunView is wrapped in a RequestStateWrapper, it won't be
+    // mounted until after latestMetrics data is available.
+    Object.values(this.props.latestMetrics).forEach((metric) => {
+      this.props.getMetricHistoryApi(this.props.runUuid, metric.key);
+    });
```

Comment:
Do you think it would be a heavy call if there are a lot of metrics? @dbczumar, is there a way to include the max and min in the latest metric itself where the BE already gets the history and computes it rather than doing this?
I see your comment [here](https://github.com/mlflow/mlflow/issues/4750#issuecomment-1068725414), but would like to understand the pros and cons of your suggestion.

Reply:
Yes this is a bit of a concern, I've made sure that fetching this data doesn't block the rest of the UI though. If the metric history is slow to query then a user might just see empty min and max values for a while until they're populated.

I have started prototyping support for storing the min and max values in the database, and am hoping that change might eventually be considered too. This is a much larger and more disruptive change though, needing DB schema and API changes (the current changes on top of this branch can bee seen at https://github.com/adamreeve/mlflow/compare/run_min_max...metric_min_max).

Reply:
@adamreeve @sunishsheth2009 Apologies for failing to consider the performance implications here earlier. As noted by @adamreeve , adjusting the service API and backend storage layer to accommodate this change is quite costly (at least a couple months of coordinated effort across multiple third-party service implementers). However, there's also a performance concern if we make too many repeated calls to GetMetricHistory on the same page. While I think this is likely fine if, e.g. a run has < 10 metrics, it may be insufficiently performant for runs with >= 10 metrics. 

One solution to this problem would be to require the user to "click to load" this information for a particular metric. 

Another solution would be to move this summary information into the individual Metric pages, which may still provide some UX benefit because, right now, the user needs to interact with the plot to fetch this information. We could also integrate this with the Metric page for run comparison: when multiple runs are being compared based on a single metric, we can include the min / max / latest metric values for each run in a table. This provides a much simpler UX relative to current one where users need to manually inspect each plot curve.

My preference would be to implement the second workflow. While it doesn't help us surface this information higher up in the MLflow UI, it still makes it significantly easier to interpret data from a single metric across one or more runs. I realize that this is substantially different from the table implemented in this PR, and I apologize again for the lack of clarity up front. Let me know what your thoughts are @adamreeve @sunishsheth2009.

Reply:
Hi @dbczumar, yes I think having this information on the individual metric pages would still be useful, and being able to show them on the comparison page would probably be more useful than just in the run page here and allow loading the history on-demand (I'm imagining a select drop-down to select from latest/min/max next to the metric name for example), so I'd be happy to take a look at implementing that.

Reply:
Ah sorry I misread your comment, I was thinking of the metric table on the run comparison page, but yes just displaying the min and max values in the single metric comparison page would make sense.